scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1614,6 +1647,40 @@ def update_cache(self, run_state):
                 local_state.cache_expiry_secs - fuzz_factor
             )
 
+    def secret(
+        self,
+        namespace,
+        name,
+        current_time=None,
+        allow_expired=False
+    ):
+        """Returns pod info for the pod specified by namespace and name or None if no pad matches.
+
+        Warning: Failure to pass current_time leads to incorrect recording of last access times, which will
+        lead to these objects being refreshed prematurely (potential source of bugs)
+
+        Querying the pod information is thread-safe, but the returned object should
+        not be written to.
+
+        @param allow_expired: If True, an object is considered present in cache even if it is expired.
+        @type allow_expired: bool
+        """
+        local_state = self._state.copy_state()
+
+        if local_state.k8s is None:
+            return
+
+        result = self._secrets_cache.lookup(
+            local_state.k8s,
+            current_time,
+            namespace,
+            name,
+            kind="Secret",
+            allow_expired=allow_expired
+        )
+        global_log.info("Secret %s/%s: %s" % (namespace, name, result.data))
```

Comment:
`result.data` will contain the actual secret value, right?

If that's indeed the case, we probably don't want to log it at all.

Reply:
Definitely, I've removed that.